OneBusAway/otpkit
# Issue 15 - saved frequent user locations

# Adding favorite locations and recent locations functionality

## Related Issues
#15 

## Description
This PR related to issue #15. Within this PR, the changes included are:
1. Adding a favorite locations section so that users can add their favorite locations.
2. If users tap a location, the location will be saved to recent locations.
3. Users can see the details of favorite locations and recent locations from the “more” button.

## Media
https://github.com/OneBusAway/otpkit/assets/74521244/52fc29e3-6a4b-4bff-b894-3136771f5bd7

## Test Instructions

To test this PR, you can do the following:

1. Tap any Origin or Destination.
2. Search for a location and tap the selected location. Check if it’s being added to recent locations. It should be visible.
3. Click the plus button in the add location section. Search for a location and tap your desired location. It should be added to the Favorite locations section.
4. If you click on the details of a favorite location, you should be able to see the details of the location’s title and subtitle.
5. Try to delete a favorite location from the favorite location details. It should be removed from the favorite locations section.

## Additional Context

Currently, this PR not having any edit functionality for favorite location yet because for MVP, I was thinking the scope of favorite locations is just marking the searched locations only
